internal/armada/scheduling/lease.go

```diff
@@ -276,22 +301,30 @@ func (c *leaseContext) leaseJobs(
 		}
 
 		candidates := make([]*api.Job, 0)
-		candidatesLimit := NewLeasePayloadLimit(limit.remainingJobCount, limit.remainingPayloadSizeLimitBytes, limit.maxExpectedJobSizeBytes)
+		candidatesLimit := newLeasePayloadLimit(limit.remainingJobCount, limit.remainingPayloadSizeLimitBytes, limit.maxExpectedJobSizeBytes)
 		candidateNodes := map[*api.Job]nodeTypeUsedResources{}
 		consumedNodeResources := nodeTypeUsedResources{}
 
 		for _, job := range topJobs {
 			requirement := common.TotalJobResourceRequest(job).AsFloat()
 			remainder = slice.DeepCopy()
 			remainder.Sub(requirement)
-			if isLargeEnough(job, c.minimumJobSize) && remainder.IsValid() && candidatesLimit.IsWithinLimit(job) {
-				newlyConsumed, ok, _ := matchAnyNodeTypeAllocation(job, c.nodeResources, consumedNodeResources)
+
+			if isJobSchedulable(c, job, remainder, candidatesLimit) {
+				newlyConsumed, ok, err := matchAnyNodeTypeAllocation(
+					job,
+					c.nodeResources,
+					consumedNodeResources,
+					c.schedulingConfig.Preemption.PriorityClasses,
+				)
 				if ok {
 					slice = remainder
 					candidates = append(candidates, job)
 					candidatesLimit.RemoveFromRemainingLimit(job)
 					candidateNodes[job] = newlyConsumed
 					consumedNodeResources.Add(newlyConsumed)
+				} else {
+					log.WithError(err).Error("failed to match node")
```

Comment:
Am I right in thinking this would mean that if a job was submitted with an invalid priority class  (e.g. because we changed priority classes between the job being submitted and it being scheduled), it would stick around for ever?

I'm tempted to say that in this case we should just submit the job as the default priority class.

Reply:
Hmmm it will go to k8s scheduler, and k8s would error saying that pod cannot be scheduled due to non-existing PC.
Will Armada catch that error and handle it? (remove it from queue)

Reply:
this is fixed